src/extensions/markdown/GPT/hooks/useAutoFocus.ts

```diff
@@ -0,0 +1,32 @@
+import {useLayoutEffect} from 'react';
+import type {RefObject} from 'react';
+
+import {focusWithoutScroll} from '../utils';
+
+export type UseAutoFocusProps = {
+    shouldFocus?: boolean;
+    containerRef?: RefObject<HTMLElement>;
+    elementSelector: string;
+};
+
+export function useAutoFocus({
```

Comment:
This hook is already in the project
Use an existing one, or update an existing one if different logic is required.

https://github.com/gravity-ui/markdown-editor/blob/main/src/react-utils/useAutoFocus.ts